frontend/src/js/util/LruCache.ts

```diff
@@ -85,6 +85,11 @@ export class LruCache<K extends string | number, V> {
   };
 
   replace = (k: K, v: V): void => {
-    this.addToCache(k, v);
+    if (this.entries[k]) {
+      this.entries[k] = v;
+      this.bumpRecency(k);
+    } else {
+      this.addToCache(k, v);
```

Comment:
let's `console.warn` a message here, since it might indicate a bug if someone is calling `replace` but the thing isn't actually in the cache already